sdks/python/apache_beam/runners/portability/prism_runner.py

```diff
@@ -0,0 +1,213 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A runner for executing portable pipelines on Apache Beam Prism."""
+
+# this will make using the list parameterized generic happy
+# on python 3.8 so we aren't revisiting this code after we
+# sunset it
+from __future__ import annotations
+
+# pytype: skip-file
+
+import logging
+import os
+import platform
+import shutil
+import stat
+import urllib
+import zipfile
+
+from urllib.error import URLError
+from urllib.request import urlopen
+
+from apache_beam.io.filesystems import FileSystems
+from apache_beam.options import pipeline_options
+from apache_beam.runners.portability import job_server
+from apache_beam.runners.portability import portable_runner
+from apache_beam.utils import subprocess_server
+from apache_beam.version import __version__ as beam_version
+
+# Prefix for constructing a download URL
+GITHUB_DOWNLOAD_PREFIX = 'https://github.com/apache/beam/releases/download/'
```

Comment:
Why? What flow is desired here? 

The logic for --prism_location already allows for arbitrary locations, if remove the hard filter that it matches this prefix, which would be a simpler way to download arbitrary code, but then it opens up using arbitrary URLs instead of signed Beam release bins.

If users want to use an arbitary binary, they don't need to host a URL for it, just build/download whatever they want, and point to the local file. I'm skeptical it's a good idea to have an unvetted user flow be able to download, unzip, and execute arbitrary files from the internet.

Longer term, I'd be delighted if we enhanced this to validate there's a sha512 file and the signature files, and check that the zips match those and are signed with Beam's signing key, to reduce likely attack vectors.

Reply:
From our conversation yesterday, it makes sense now to have the const here.